### DISCUSSION TITLE
ConsumerControl forward/backward not working

### DISCUSSION THREAD

**Discussion by HasseM** [adafruit/Adafruit_TinyUSB_Arduino#266] [Q&A]
I have it all working on an RPi Pico:

To start/stop media playback:
usb_hid.sendReport16(RID_CONSUMER_CONTROL, HID_USAGE_CONSUMER_PLAY_PAUSE);

But now I also want to fastforward/backward in a video. However, there is no HID_USAGE_CONSUMER_XXXX for this. I tried 0x003B:
usb_hid.sendReport16(RID_CONSUMER_CONTROL, 0x003B);
But that didnt work either.

Anyone has a tip on how to go forward/backward in a video?

**Comment by hathach** [comment]
try sending right and left arrow from normal keyboard, it should probably work ?

**Reply by HasseM** [reply]
The nice thing about these media keys is that they also work when the browser is not the active window. Arrow keys only work on the active window. My window is not the active window. So therefor I need the media keys.

**Reply by hathach** [reply]
have you checked HID usage table specs for consumer control

**Reply by HasseM** [reply]
Yes, but I am unsure about which one I should use for ffwd/bwd. It is defenitely not in the library. But hardcoding it on 0x003B does not do anything.

**Reply by hathach** [reply]
you could write the sketch to take input from serial and send key press/released for 1 second. That's way it is easier to test with. I haven't revised the consumer key symbol for some time, it may lack behind the specs, check the HID spec or google/chatGPT for detail, or just try it out several keycode that look promising.

**Reply by HasseM** [reply]
I did a counter starting at B0, going upwards at each key press (sending one message, followed by 0 after 2ms). I Used this page for testing: https://googlechrome.github.io/samples/media-session/video.html
But no forward/backward messages in the javascript output there. Even not with it being the active window (just to be sure).
So I was wondering of OOC messages should be in a different format or so. I'm no where near a HID expert (yet).

**Comment by HasseM** [comment]
I am also wondering if it may have to do with OSC (One Shot Control) vs OOC (On Off Control) because the OSC ones work......
